cmd/node-controller-manager/app/controllermanager.go

```diff
@@ -177,20 +192,24 @@ func StartControllers(s *options.NCMServer,
 	if availableResources[awsGVR] || availableResources[azureGVR] || availableResources[gcpGVR] {
```

Comment:
availableResources[] checks for CRDs in shoot cluster currently. What we'll need is a kubeClient that points to the seed cluster's coreV1 objects ?

Reply:
Yes, this is interesting. 
We should ideally have 3 internal clients, controlCoreClient, controlMachineClient , targetCoreClient. We would need to access `secrets`, `crd` via controlCoreClient.